seqr/views/utils/dataset_utils.py

```diff
@@ -356,55 +326,76 @@ def _load_rna_seq_file(file_path, user, mapping_file, parse_row, expected_column
     samples_by_id = defaultdict(dict)
     f = file_iter(file_path, user=user)
     header = _parse_tsv_row(next(f))
-    missing_cols = set(expected_columns) - set(header)
+    required_column_map = {
+        column_map.get(col, col): col for col in [SAMPLE_ID_COL, PROJECT_COL, GENE_ID_COL, TISSUE_COL]
+    }
+    expected_cols = set(column_map.values())
+    expected_cols.update(required_column_map.keys())
+    missing_cols = expected_cols - set(header)
     if missing_cols:
         raise ValueError(f'Invalid file: missing column(s): {", ".join(sorted(missing_cols))}')
 
     sample_id_to_tissue_type = {}
     samples_with_conflict_tissues = defaultdict(set)
     errors = []
+    missing_required_fields = defaultdict(list)
     gene_ids = set()
-    missing_gene_samples = set()
     for line in tqdm(f, unit=' rows'):
         row = dict(zip(header, _parse_tsv_row(line)))
-        for sample_id, row_dict in parse_row(row):
-            tissue_type = TISSUE_TYPE_MAP.get(row_dict.pop(TISSUE_COL, None))
```

Comment:
a lof of this change is just unindenting, since theres no longer `parse_row` iterator with a for loop and instead just a one-to-one parsing of each row